src/services/edusharing/logic/connector.js

```diff
@@ -0,0 +1,165 @@
+		this.accessToken = null; // ACCESSTOKEN
+		EduSharingConnector.instance = this;
+	}
+
+	static get headers() {
+		return {
+			Accept: 'application/json',
+			'Content-type': 'application/json'
+		};
+	}
+
+	static get authorization() {
+		const userName = process.env.ES_USER;
+		const pw = process.env.ES_PASSWORD;
+		const headers = Object.assign({}, EduSharingConnector.headers, {
+			Authorization: `Basic ${Buffer.from(`${userName}:${pw}`).toString(
+				'base64'
+			)}`
+		});
+		return headers;
+	}
+
+	// gets cookie (JSESSION) and attach it to header
+	getCookie() {
+		const cookieOptions = {
+			uri: `${process.env.ES_DOMAIN}/edu-sharing/rest/authentication/v1/validateSession`,
+			method: 'GET',
+			headers: EduSharingConnector.authorization,
+			resolveWithFullResponse: true,
+			json: true
+		};
+		return request(cookieOptions)
+			.then(result => {
+				if (
+					result.statusCode !== 200 ||
+					result.body.isValidLogin !== true
+				) {
+					throw Error('authentication error with edu sharing');
+				}
+				return result.headers['set-cookie'][0];
+			})
+			.catch(err => {
+				// eslint-disable-next-line no-console
+				console.error('error: ', err);
+			});
+	}
+	// gets access_token and refresh_token
+	getAuth() {
+		const oauthoptions = {
+			method: 'POST',
+			url: `${process.env.ES_DOMAIN}/edu-sharing/oauth2/token`,
+			headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
+			body: `grant_type=${'password'}&client_id=${process.env.ES_CLIENT_ID}&client_secret=${process.env.ES_OAUTH_SECRET}&username=${process.env.ES_USER}&password=${process.env.ES_PASSWORD}`,
+			timeout: REQUEST_TIMEOUT
+		};
+		return request(oauthoptions).then(result => {
+			if (result) {
+				const parsedResult = JSON.parse(result);
+				return parsedResult.access_token;
+			}
+			console.error('Oauth failed');
+			return null;
+		});
+	}
+
+	checkEnv() {
+
+		return (
+			process.env.ES_DOMAIN &&
+			process.env.ES_USER &&
+			process.env.ES_PASSWORD &&
+			process.env.ES_GRANT_TYPE &&
+			process.env.ES_OAUTH_SECRET &&
+			process.env.ES_CLIENT_ID
+		);
+	}
+
+	async login() {
+		this.authorization = await this.getCookie();
+		this.accessToken = await this.getAuth();
+	}
+
+	isLoggedin() {
+		// returns false if cookie or accesstoken is falsy
+		return !!this.authorization && !!this.accessToken;
+	}
+
+	async GET(data) {
+
+		const limit = data.query.$limit || 9;
+		const skip = data.query.$skip || 0;
+		const searchWord = data.query.searchWord || 'img'; // will give pictures of flowers as default
+		const contentType = data.query.contentType || 'FILES'; // enum: ['FILES', FILES_AND_FOLDERS,COLLECTIONS, TOOLPERMISSIONS, ALL]
+
+		if (!this.checkEnv()) {
+			return 'Update your env variables. See --> src/services/edusharing/envTemplate';
+		}
+
+		if (this.isLoggedin() === false) {
+			await this.login();
+		}
+		const options = {
+			method: 'POST',
+			// This will be changed later with a qs where sorting, filtering etc is present. 
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Trailing spaces not allowed.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4791331)